config.go

```diff
@@ -4,17 +4,31 @@ import (
 	"errors"
 	"fmt"
 	"regexp"
+	"sync"
 )
 
 type Config struct {
 	RequiredAlias        map[string]string
 	Rules                []*Rule
 	DisallowUnaliased    bool
 	DisallowExtraAliases bool
+	muRules              sync.RWMutex
 }
 
 func (c *Config) CompileRegexp() error {
-	rules := make([]*Rule, 0, len(c.RequiredAlias))
+	c.muRules.RLock()
+	rules := c.Rules
+	c.muRules.RUnlock()
+	if rules != nil {
+		return nil
+	}
+	c.muRules.Lock()
+	defer c.muRules.Unlock()
+	// Check if another process got here first.
+	if c.Rules != nil {
+		return nil
+	}
```

Comment:
```suggestion
	c.muRules.Lock()
	defer c.muRules.Unlock()
	// Check if another process got here first.
	if c.Rules != nil {
		return nil
	}
```

maybe you can simplify this section?

Reply:
The RLock allows readers to not block each other though likely won't make any real difference in this case since it's just checking the variable. I can make this change but will also then change the RWMutex to just Mutex for clarity.

Reply:
I made the change to simplify the locks to use  Mutex instead of RWMutex. `findRule` takes a local reference to the rules so it doesn't need to hold a lock while doing the matching. This is safe since the rules variable is overwritten and the slice is never modified.